core/bin/sqlite_worker.rs

```diff
@@ -144,6 +171,123 @@ async fn db_query(
     }
 }
 
+#[derive(serde::Deserialize)]
+struct DatabasesRowsUpsertPayload {
+    rows: Vec<DatabaseRow>,
+    truncate: Option<bool>,
+}
+
+async fn databases_rows_upsert(
+    extract::Path((database_id, table_id)): extract::Path<(String, String)>,
+    extract::Json(payload): extract::Json<DatabasesRowsUpsertPayload>,
+    Extension(state): Extension<Arc<WorkerState>>,
+) -> (StatusCode, Json<APIResponse>) {
+    // Terminate the running DB thread if it exists.
```

Comment:
Note: this can be improved in the future by issuing the right SQL to the running DB. For now, we'll just kill it if it's running.